File: src/main/java/czopekonieszczuk/scalatomato/activities/LoginActivity.scala
package czopekonieszczuk.scalatomato.activities

import android.util.Log
import czopekonieszczuk.scalatomato.R
import czopekonieszczuk.scalatomato.databases._
import org.scaloid.common._

class LoginActivity extends SActivity {

  onCreate {

    contentView = new SScrollView {
      this += new SVerticalLayout {
        val appNameTextView = STextView(R.string.app_name).textSize(30 dip).<<.marginBottom(50 dip).>>
        val loginTextView = STextView(R.string.login_text).<<.marginBottom(20 dip).>>
        val loginEditText = SEditText().<<.marginBottom(20 dip).>>
        val passwordTextView = STextView(R.string.password_text).<<.marginBottom(20 dip).>>
        val passwordEditText = SEditText().<<.marginBottom(20 dip).>> inputType TEXT_PASSWORD
        Log.d("LoginActivity.onCreate", "Created TextViews and EditTexts")

        this += new SLinearLayout {
          SButton(R.string.login_text).<<.Weight(1.0f).>>.onClick(loginUserToApp(loginEditText.getText.toString, passwordEditText.getText.toString))
          SButton(R.string.register_text).<<.Weight(1.0f).>>.onClick(startRegisterActivity)
          Log.d("LoginActivity.onCreate", "Created login and register buttons")
        }

        //TEST BUTTONS
//        val testButton1 = SButton("Add test/123").onClick(testAddUser)
//        val testButton2 = SButton("Delete test/123").onClick(testDeleteUser)
//        val testButton3 = SButton("Tomato activity").onClick(testTomatoUser)
//        val testButton4 = SButton("List tomatoes").onClick(testUserTomatoes)
//        Log.d("LoginActivity.onCreate", "Created test buttons")
      }.padding(20.dip)
    }

  }


  def testAddUser {
    val user = new User("test", "123")
    val udb = new UserDatabaseHelper(this)
    try {
      udb.addUser(user)
      Log.d("LoginActivity.testAddUser", "Added test user")
    } catch {
      case exist: android.database.sqlite.SQLiteConstraintException => toast("Uzytkownik istnieje!")
    }
  }

  def testDeleteUser() {
    val udb = new UserDatabaseHelper(this)
    udb.deleteUser(4)
    udb.deleteUser(3)
    udb.deleteUser(2)
    udb.deleteUser(1)
    longToast("Usunieto uzytkownikow")
    Log.d("LoginActivity.testDeleteUser", "Deleted test users")
  }

  def testTomatoUser {
    val userId: Long = 5
    val intent = SIntent[TomatoActivity]
    intent.putExtra("userId", userId)
    Log.d("LoginActivity.testTomatoUser", "Put to TomatoActivity intent userId: " +userId.toString)
    startActivity(intent)
    Log.d("LoginActivity.testTomatoUser", "Started TomatoActivity")
  }

  def testUserTomatoes {
    val userId: Long = 5
    val intent = SIntent[UserTomatoesActivity]
    intent.putExtra("userId", userId)
    Log.d("LoginActivity.testUserTomatoes", "Put to TomatoActivity intent userId: " +userId.toString)
    startActivity(intent)
    Log.d("LoginActivity.testUserTomatoes", "Started UserTomatoesActivity")
  }

  def loginUserToApp(login: String, password: String) {
    val udb = new UserDatabaseHelper(this)
    val userId: Long = udb.loginUser(login, password)
    if(userId == -1) {
      new AlertDialogBuilder(R.string.failed, R.string.login_failed_text) {
        neutralButton()
      }.show()
      Log.d("LoginActivity.loginUserToApp", "Created Alert, not found user")
    } else {
      Log.d("LoginActivity.loginUserToApp", "Logged userId: "+userId.toString)
      val intent = SIntent[UserActivity]
      intent.putExtra("userId", userId)
      Log.d("LoginActivity.loginUserToApp", "Put to intent userId: " +userId.toString)
      startActivity(intent)
      Log.d("LoginActivity.loginUserToApp", "Started UserActivity")
    }
  }

  def startRegisterActivity {
    val intent = SIntent[RegisterActivity]
    startActivity(intent)
    Log.d("LoginActivity.startRegisterActivity", "Started RegisterActivity")
  }
}

File: src/main/java/czopekonieszczuk/scalatomato/javaxml/LoginActivity.java
package czopekonieszczuk.scalatomato.javaxml;


import android.app.AlertDialog;
import android.app.Dialog;
import android.app.DialogFragment;
import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;
import android.widget.EditText;

import czopekonieszczuk.scalatomato.R;
import czopekonieszczuk.scalatomato.activities.RegisterActivity;
import czopekonieszczuk.scalatomato.activities.UserActivity;
import czopekonieszczuk.scalatomato.databases.UserDatabaseHelper;

/**
 * Created by czopo on 1/6/16.
 */
public class LoginActivity extends AppCompatActivity {

    EditText loginEditText;
    EditText passwordEditText;
    Button loginButton;
    Button registerButton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_login);

        loginEditText = (EditText) findViewById(R.id.loginEditText);
        passwordEditText = (EditText) findViewById(R.id.passwordEditText);
        loginButton = (Button) findViewById(R.id.loginButton);
        registerButton = (Button) findViewById(R.id.registerButton);

        Log.d("LoginActivity.onCreate", "Created TextViews and EditTexts");

        loginButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                UserDatabaseHelper udb = new UserDatabaseHelper();
                String login = loginEditText.getText().toString();
                String password = passwordEditText.getText().toString();
                Long userId = udb.loginUser(login, password);
                if (userId == -1) {
                    new DialogFragment() {
                        @Override
                        public Dialog onCreateDialog(Bundle savedInstanceState) {
                            AlertDialog.Builder builder = new AlertDialog.Builder(getActivity());
                            builder.setTitle(R.string.failed);
                            builder.setMessage(R.string.login_failed_text);
                            return builder.create();
                        }
                    }.show(getFragmentManager(), null);
                    Log.d("LoginActivity.loginUserToApp", "Created Alert, not found user");
                } else {
                    Log.d("LoginActivity.loginUserToApp", "Logged userId: " + userId);
                    Intent intent = new Intent(getApplicationContext(), UserActivity.class);
                    intent.putExtra("userId", userId);
                    Log.d("LoginActivity.loginUserToApp", "Put to intent userId: " + userId);
                    startActivity(intent);
                    Log.d("LoginActivity.loginUserToApp", "Started UserActivity");
                }
            }
        });

        registerButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent intent = new Intent(getApplicationContext(), RegisterActivity.class);
                startActivity(intent);
                Log.d("LoginActivity.startRegisterActivity", "Started RegisterActivity");
            }
        });

        Log.d("LoginActivity.onCreate", "Created login and register buttons");

    }
}
